PhoneGap/SwrvePlugin/platforms/ios/CDVAppDelegate+Swizzling.m

```diff
@@ -0,0 +1,52 @@
+#import <objc/runtime.h>
+#import "AppDelegate.h"
+#import "Swrve.h"
+#import "SwrvePlugin.h"
+
+static void swizzleMethod(Class class, SEL destinationSelector, SEL sourceSelector);
+
+@implementation CDVAppDelegate (Swrve)
+
++ (void)load {
+    swizzleMethod([CDVAppDelegate class],
+        @selector(application:didFinishLaunchingWithOptions:),
+        @selector(my_application:didFinishLaunchingWithOptions:));
+
+    swizzleMethod([CDVAppDelegate class],
+        @selector(application:didReceiveRemoteNotification:),
+        @selector(my_application:didReceiveRemoteNotification:));
+}
+
+- (BOOL)my_application: (UIApplication*)application
+            didFinishLaunchingWithOptions:(NSDictionary*)launchOptions {
+    NSNumber *appId = NSBundle.mainBundle.infoDictionary[@"AppId"];
+    NSString *apiKey = NSBundle.mainBundle.infoDictionary[@"ApiKey"];
+    SwrveConfig* config = [[SwrveConfig alloc] init];
+    [SwrvePlugin initWithAppID:[appId intValue]
+                        apiKey:apiKey
+                        config:config
+                viewController:self.viewController
+                launchOptions:launchOptions];
+
+     return [self my_application:application didFinishLaunchingWithOptions:launchOptions];
+}
+
+-(void) my_application: (UIApplication*) application didReceiveRemoteNotification:(NSDictionary *)userInfo {
+    [SwrvePlugin application:application didReceiveRemoteNotification:userInfo];
+    [self my_application:application didReceiveRemoteNotification:userInfo];
+}
+@end
+
+#pragma mark Swizzling
+
+static void swizzleMethod(Class class, SEL destinationSelector, SEL sourceSelector) {
+    Method destinationMethod = class_getInstanceMethod(class, destinationSelector);
```

Comment:
We had a bug while back where we needed to check up in the hierarchy not just the class provided in our native SDK and it seems like it might be present in these changes too. The method could be implemented in a parent class and this method would not find it.